File: extras/participacion4.py
import math

class Circulo2D:
    def __init__(self, x=0, y=0, radio=1):
        self.x = x
        self.y = y
        self.radio = radio

    def get_x(self):
        return self.x

    def get_y(self):
        return self.y

    def get_radio(self):
        return self.radio

    def get_area(self):
        return math.pi * self.radio ** 2

    def get_perimetro(self):
        return 2 * math.pi * self.radio

    def contiene(self, x, y):
        distancia = math.sqrt((self.x - x) ** 2 + (self.y - y) ** 2)
        return distancia <= self.radio

    def contieneCirculo(self, circulo):
        distancia = math.sqrt((self.x - circulo.x) ** 2 + (self.y - circulo.y) ** 2)
        return distancia + circulo.radio <= self.radio

    def sobrepone(self, circulo) :
        distancia = math.sqrt((self.x - circulo.x) ** 2 + (self.y - circulo.y) ** 2)
        return distancia < self.radio + circulo.radio

    def mostrar(self):
        print(f"Area: {self.get_area():.2f}")
        print(f"Perimetro: {self.get_perimetro():.2f}")
        print(self.contiene(2.5, 0))
        print(self.contieneCirculo(Circulo2D(2, 0, 0.5)))
        print(self.sobrepone(Circulo2D(0, 0, 2)))


if __name__ == "__main__":
    c1 = Circulo2D(2, 0, 1)
    c1.mostrar()
    

File: extras/participacion4.java
package pack1;

public class Circulo2D {
    private double x;
    private double y;
    private double radio;

    public Circulo2D() {
        this(0, 0, 1);
    }

    public Circulo2D(double x, double y, double radio) {
        this.x = x;
        this.y = y;
        this.radio = radio;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public double getRadio() {
        return radio;
    }

    public double getArea() {
        return Math.PI * radio * radio;
    }

    public double getPerimetro() {
        return 2 * Math.PI * radio;
    }

    public boolean contiene(double x, double y) {
        double distancia = Math.sqrt(Math.pow(this.x - x, 2) + Math.pow(this.y - y, 2));
        return distancia <= radio ;
    }

    public boolean contieneCirculo(Circulo2D circulo) {
        double distancia = Math.sqrt(Math.pow(this.x - circulo.x, 2) + Math.pow(this.y - circulo.y, 2));
        return distancia + circulo.radio <= this.radio;
    }

    public boolean sobrepone(Circulo2D circulo) {
        double distancia = Math.sqrt(Math.pow(this.x - circulo.x, 2) + Math.pow(this.y - circulo.y, 2));
        return distancia < this.radio + circulo.radio;
    }

    public static void main(String[] args) {
        Circulo2D c1 = new Circulo2D(2, 0, 1);
        System.out.printf("Area: %.2f%n", c1.getArea());
        System.out.printf("Perimetro: %.2f%n", c1.getPerimetro());
        System.out.println(c1.contiene(2.5, 0));
        System.out.println(c1.contieneCirculo(new Circulo2D(2, 0, 0.5)));
        System.out.println(c1.sobrepone(new Circulo2D(0, 0, 2)));
    }
}


